web/client/api/catalog/COG.js

```diff
@@ -57,8 +58,23 @@ export const getProjectionFromGeoKeys = (image) => {
 
     return null;
 };
+const abortError = (reject) => reject(new DOMException("Aborted", "AbortError"));
+/**
+ * fromUrl with abort fetching of data and data slices
+ */
+const fromUrl = (url, signal) => {
+    if (signal?.aborted) {
+        return abortError(Promise.reject);
+    }
+    return new Promise((resolve, reject) => {
+        signal?.addEventListener("abort", () => abortError(reject));
+        return fromGeotiffUrl(url)
+            .then((image)=> image.getImage()) // Fetch and read first image to get medatadata of the tif
+            .then((image) => resolve(image))
```

Comment:
Basically, as discussed with @dsuren1 , we can canel the pending task to make the user continue operating if the "Save" takes too long, but we can not cancel the effective requests pending 

This can cause the pending requests to be blocking and slowing down the load on that particular domain in any case (because of bandwitdh and max number of parallel requests allowed to a specific domain by the browser etc...)

Please report the known issue notice, or we loose memory of it. 
Having an issue about it linking could be useful to track the problem too. 
@tdipisa should we open it ? 

I ask @dsuren1 to link the proper issue to github repo of geotiffjs, for completeness. 
```suggestion
            .then((image) => resolve(image)) // note: the cacellation will not cancel the request because of issue in
```

Reply:
Updated note on the function and created an issue in geotiff.js